colossus/src/main/scala/colossus/service/ServiceClient.scala

```diff
@@ -296,4 +300,13 @@ extends Controller[O,I](codec, ControllerConfig(config.pendingBufferSize, config
     }
   }
 
+
+  override def idleCheck(period: Duration) {
+    super.idleCheck(period)
+
+    if (sentBuffer.size > 0 && sentBuffer.front.isTimedOut(System.currentTimeMillis)) {
+      // the oldest sent message has expired with no response - shutdown the connection
+      disconnect()
```

Comment:
That first thing might be a bug.  Calling disconnect() should always be a "final" operation as it is intended to only be used when the user actually wants to shut things down.  I will check on this and maybe write some more tests.

So if we have this go through the same process that `connectionLost` goes though, the client will deal with all pending and in-flight requests correctly, since what happens depends on whether `failFast` is enabled or not.